src/mpid/ch4/src/ch4_coll_impl.h

```diff
@@ -83,6 +83,65 @@
 
 #define MPIR_ALLREDUCE_SHM_PER_LEADER_MAX 4194304
 
+static void MPIDI_Coll_host_buffer_genq_alloc(const void *sendbuf, const void *recvbuf,
+                                              MPI_Aint count, MPI_Datatype datatype,
+                                              void **host_sendbuf, void **host_recvbuf,
+                                              MPL_pointer_attr_t send_attr,
+                                              MPL_pointer_attr_t recv_attr, MPI_Aint shift)
+{
+    void *tmp_send, *tmp_recv;
+    if (sendbuf != MPI_IN_PLACE) {
+        MPIDU_genq_private_pool_alloc_cell(MPIDI_global.gpu_coll_pool, (void **) &tmp_send);
```

Comment:
Just pointing out that since the private pool has a maximum number of buffers, it is possible for this cell allocation to fail. There probably needs to be a fallback for when the pool is empty.

Reply:
In this PR, we are doing this for blocking collectives. So it should get a buffer from the pool.

Reply:
`MPIDU_genq_private_pool_alloc_cell` can fail. There is no progress call in it to ensure a buffer is available. This code assumes the buffer is always there after return which is not always true.

Reply:
Added `MPIR_Assert` calls based on other usage of `MPIDU_genq_private_pool_alloc_cell` to ensure the buffer is available

Reply:
Can you put a FIXME comment in to do graceful fallback? I can add it after the fact.

Reply:
Added